ql/src/test/results/clientpositive/llap/explain_locks.q.out

```diff
@@ -233,20 +221,14 @@ POSTHOOK: Input: default@target@p=2/q=2
 POSTHOOK: Output: default@merge_tmp_table
 POSTHOOK: Output: default@target
 POSTHOOK: Output: default@target@p=1/q=2
-POSTHOOK: Output: default@target@p=1/q=2
-POSTHOOK: Output: default@target@p=1/q=3
 POSTHOOK: Output: default@target@p=1/q=3
 POSTHOOK: Output: default@target@p=2/q=2
-POSTHOOK: Output: default@target@p=2/q=2
 LOCK INFORMATION:
 default.source -> SHARED_READ
 default.target.p=1/q=2 -> SHARED_READ
 default.target.p=1/q=3 -> SHARED_READ
 default.target.p=2/q=2 -> SHARED_READ
 default.target.p=2/q=2 -> SHARED_WRITE
-default.target.p=2/q=2 -> SHARED_WRITE
```

Comment:
I wonder why were these lock duplicated?

Reply:
If `hive.merge.split.update` is `false` the merge statement has three type of outputs: `insert`, `delete`, `update` the same partitioned table.

`RewriteSemanticAnalyzer.updateOutputs()` adds the affected partitions twice: one set for `delete` and one for `update`
https://github.com/apache/hive/blob/6f29c1acf04fa1f1b84a6fb86f92bdc65b584567/ql/src/java/org/apache/hadoop/hive/ql/parse/RewriteSemanticAnalyzer.java#L339

When `hive.merge.split.update` is `true` only `insert` and `delete` outputs are present so partitions are listed only once.

Reply:
thank you for the explanation